src/Mod/Part/Gui/SectionCutting.cpp

```diff
@@ -0,0 +1,1475 @@
+                break;
+            }
+        }
+        doc->removeObject(BoxZName);
+    }
+    if (doc->getObject(CutYName)) {
+        anObject = doc->getObject(CutYName);
+        for (it = ObjectsListVisible.begin(); it != ObjectsListVisible.end(); ++it) {
+            if ((*it) == anObject) {
+                ObjectsListVisible.erase((it));
+                break;
+            }
+        }
+        doc->removeObject(CutYName);
+    }
+    if (doc->getObject(BoxYName)) {
+        anObject = doc->getObject(BoxYName);
+        for (it = ObjectsListVisible.begin(); it != ObjectsListVisible.end(); ++it) {
+            if ((*it) == anObject) {
+                ObjectsListVisible.erase((it));
+                break;
+            }
+        }
+        doc->removeObject(BoxYName);
+    }
+    if (doc->getObject(CutXName)) {
+        anObject = doc->getObject(CutXName);
+        for (it = ObjectsListVisible.begin(); it != ObjectsListVisible.end(); ++it) {
+            if ((*it) == anObject) {
+                ObjectsListVisible.erase((it));
+                break;
+            }
+        }
+        doc->removeObject(CutXName);
+    }
+    if (doc->getObject(BoxXName)) {
+        anObject = doc->getObject(BoxXName);
+        for (it = ObjectsListVisible.begin(); it != ObjectsListVisible.end(); ++it) {
+            if ((*it) == anObject) {
+                ObjectsListVisible.erase((it));
+                break;
+            }
+        }
+        doc->removeObject(BoxXName);
+    }
+    if (doc->getObject(CompoundName)) {
+        auto compoundObject = doc->getObject(CompoundName);
+        Part::Compound* pcCompoundDel = static_cast<Part::Compound*>(compoundObject);
+        std::vector<App::DocumentObject*> compoundObjects;
+        pcCompoundDel->Links.getLinks(compoundObjects);
+        // first delete the compound
+        for (it = ObjectsListVisible.begin(); it != ObjectsListVisible.end(); ++it) {
+            if ((*it) == anObject) {
+                ObjectsListVisible.erase((it));
+                break;
+            }
+        }
+        doc->removeObject(CompoundName);
+        // now delete the objects that have been part of the compound
+        for (it = compoundObjects.begin(); it != compoundObjects.end(); it++) {
+            for (auto itOV = ObjectsListVisible.begin(); itOV != ObjectsListVisible.end(); ++itOV) {
+                if ((*itOV) == doc->getObject((*it)->getNameInDocument())) {
+                    ObjectsListVisible.erase((itOV));
+                    break;
+                }
+            }
+            doc->removeObject((*it)->getNameInDocument());
+        }
+    }
+
+    // make all objects visible that have been visible when the dialog was called
+    // because we made them invisible when we created cuts
+    for (it = ObjectsListVisible.begin(); it != ObjectsListVisible.end(); ++it) {
+        if ((*it)->isValid()) // a formerly visible object might have been deleted

```

Comment:
To understand: How can there be dangling pointers? I mean I just created this list few lines above. And on creation time I checked that the objects exist. Even if the users wants, he cannot have deleted the objects within the few milliseconds between the list creation and checking.

Nevertheless I can change to App::DocumentObjectT .

Reply:
You fill the list ObjectsListVisible in the constructor of SectionCut. But between the constructor call and this function call a lot of things can happen because the dialog is non-modal and thus allows the user to modify the document in the meantime.

Assume you added three objects to ObjectsListVisible and the user has deleted one of them before continuing the interaction with the dialog. Now the dialog class has nothing implemented to be notified about document changes and thus won't update ObjectsListVisible accordingly.

Reply:
done in #6360